NLog.Windows.Forms/RichTextBoxTarget.cs

```diff
@@ -299,6 +318,107 @@ public RichTextBoxTargetMessageRetentionStrategy MessageRetention
         private volatile Queue<MessageInfo> messageQueue = null;
 
         /// <summary>
+        /// If set to true, using "rtb-link" renderer (<see cref="RichTextBoxLinkLayoutRenderer"/>) would create clickable links in the control.
+        /// <seealso cref="LinkClicked"/>
+        /// </summary>
+        [DefaultValue(false)]
+        public bool SupportLinks 
+        {
+            get { return supportLinks; }
+            set
+            {
+                supportLinks = value;
+                if (supportLinks)
+                {
+                    lock (linkRegexLock)
+                    {
+                        if (linkAddRegex == null)
```

Comment:
Well, I don't like turning `linkAddRegex` and `linkRemoveRtfRegex` into properties with lazy initialization. It wold look like that, right:

```c#
private static Regex LinkAddRegex 
{
   get 
   {
       lock (linkRegexLock) 
       {
           if (linkAddRegex == null)
           {
              linkAddRegex = new Regex(...);
           }
       }
       return linkAddRegex;
   }
}
```

So we would have to not only check if the regex actually exists, but also synchronize this check. And it would be inter-instance synchronization because of `static` fields nature. And this would happen for every log event.

Current code guards access to these two regexes with `if (SupportLinks)` which assures creation of regexes. (I'll update the code to assure the underlying `supportLinks` field is set after the regex initialization.) So all the locking happens on rarely called property change.

I also don't like (costy) regex initialization to happen during actual log event, so I would like to keep initialization in the `SupportLinks` setter anyway, which lessens the value of having lazily-initialized properties.

Maybe we could just add better comments to this regexes usage? Or remove lazy initialization as whole?